src/Integration.UnitTests/Binding/BindingWorkflowTests.cs

```diff
@@ -365,6 +407,43 @@ public void BindingWorkflow_DiscoverProjects_OutputsExcludedProjects()
             var progressEvents = new ConfigurableProgressStepExecutionEvents();
 
             List<Project> projects = new List<Project>();
+            for (int i = 0; i < 2; i++)
```

Comment:
All these tests seem to share a lot in common. Can we extract the copy-pasted parts to dedicated methods?